RInAppMessaging/Classes/CampaignDispatcher.swift

```diff
@@ -89,10 +89,26 @@ internal class CampaignDispatcher: CampaignDispatcherType, TaskSchedulable {
             return
         }
 
+        // fetch image from url and display if successful
+        if let resImgUrlString = campaign.data.messagePayload.resource.imageUrl, let resImgUrl = URL(string: resImgUrlString) {
```

Comment:
I'm wondering if adding additional concurrency here won't cause any unwanted side effects (like race conditions) now or in the future. Should we make image downloading synchronous? Is it worth it?

Reply:
Should be fine if I dispatch the URLSession's completion block the on same queue as where the campaignRepo is being mutated/decremented. Was initially concerned about `CampaignDispatcher` stalling due to `Data(contentsOf:)` long 60s(?) default timeout and slow/intermittent connections

Reply:
Ok so let's leave as it is. Yeah, the previous implementation wasn't good in terms of error and timeout handling.